File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.py
class Solution:
    def subarraySum(self, nums: list[int], k: int) -> int:
        counter = 0
        n = len(nums)
        prefix_sums = {0: 1}
        current_sum = 0
        
        for i in range(n):
            current_sum += nums[i]
            if (current_sum - k) in prefix_sums:
                counter += prefix_sums[current_sum - k]
            prefix_sums[current_sum] = prefix_sums.get(current_sum, 0) + 1
            
        return counter

File: Data Structures & Algorithms/jump-game/submission-1.cpp
class Solution{
public:

    bool canJump(vector<int>& nums){
        int sum = nums.size()-1;

        for(int i = nums.size()-2;i>=0;i--){
            if(i+nums[i] >=  sum) sum = i;
        }
        return (sum == 0);
    }

};